pkg/kudoctl/resources/install/operator.go

```diff
@@ -67,18 +66,6 @@ func OperatorAndOperatorVersion(
 }
 
 func installDependencies(client *kudo.Client, ov *kudoapi.OperatorVersion, dependencies []deps.Dependency) error {
-	// The KUDO controller will create Instances for the dependencies. For this
```

Comment:
While working on this PR, I realized that updating the tasks here (in L80) had a bug, now that dependencies are relative to their parent. Consider a folder structure like:
```
.
└── child   # first child
│   └── child   # second child
│   │   └── operator.yaml
│   └── operator.yaml
└── operator.yaml
```
Now, the parent references the first child as `./child` and it references the second child as `./child` too. The `updateKudoOperatorTasks` method below would search the dependencies by the `dependency.PackageName`  and update the corresponding task but with two identical `PackageNames` this would lead to bugs. 

I've moved this logic in the `resolve.go::dependencyWalk` method. Now each task is updated when the dependency is resolved (which is arguably a much better place to do that).

Reply:
This setup would lead to issues anyway, right? Because the operatorName for both children is `child`? It may work if they have different appVersion/operatorVersion, but if those are the same too, the generated name would clash on installation of the OV.

Reply:
`./child` is the *folder* name, the operator name is known *after* it is resolved. And the folders can be called whatever 🤷

Reply:
Ah, right. Yeah, if the operator names are different then it should work. Yeah 👍